products/BellHybrid/apps/application-bell-relaxation/model/RelaxationSongsRepository.cpp

```diff
@@ -0,0 +1,135 @@
+                                                         const std::map<MusicType, std::string> &pathPrefixes)
+        : app::AsyncCallbackReceiver{application}, application{application}, pathPrefixes{pathPrefixes}
+    {}
+
+    void RelaxationSongsRepository::init()
+    {
+        for (const auto &[musicType, musicPath] : pathPrefixes) {
+            updateFilesCount(musicType, musicPath);
+        }
+    }
+
+    void RelaxationSongsRepository::updateFilesCount(const MusicType &type, const std::string &path)
+    {
+        auto query = std::make_unique<db::multimedia_files::query::GetCountForPath>(path);
+        auto task  = app::AsyncQuery::createFromQuery(std::move(query), db::Interface::Name::MultimediaFiles);
+        task->setCallback([this, &type](auto response) {
+            auto result = dynamic_cast<db::multimedia_files::query::GetCountResult *>(response);
+            if (result == nullptr) {
+                return false;
+            }
+            return updateCountCallback(type, result->getCount());
+        });
+        task->execute(application, this);
+    }
+
+    bool RelaxationSongsRepository::updateCountCallback(const MusicType &type, std::uint32_t count)
+    {
+        filesPerType[type] = count;
+        return true;
+    }
+
+    void RelaxationSongsRepository::getMusicFiles(
+        const MusicType &musicType,
+        std::uint32_t offset,
+        std::uint32_t limit,
+        const RelaxationSongsRepository::OnGetMusicFilesListCallback &viewUpdateCallback)
+    {
+        auto taskCallback = [this, viewUpdateCallback, offset, musicType](auto response) {
+            auto result = dynamic_cast<db::multimedia_files::query::GetLimitedResult *>(response);
+            if (result == nullptr) {
+                return false;
+            }
+            for (auto &record : result->getResult()) {
+                musicFilesViewCache.records.push_back(record);
+            }
+            musicFilesViewCache.recordsOffset = offset;
+            musicFilesViewCache.recordsCount  = getRecordsCount();
+
+            if (viewUpdateCallback) {
+                viewUpdateCallback(musicFilesViewCache.records, musicFilesViewCache.recordsCount);
+            }
+            updateCountCallback(musicType, result->getCount());
+            return true;
+        };
+
+        auto query = std::make_unique<db::multimedia_files::query::GetLimitedByPaths>(
+            std::vector<std::string>{pathPrefixes[musicType]}, offset, limit);
+        auto task = app::AsyncQuery::createFromQuery(std::move(query), db::Interface::Name::MultimediaFiles);
+        task->setCallback(taskCallback);
+        task->execute(application, this);
+    }
+
+    std::uint32_t RelaxationSongsRepository::getRecordsCount()
+    {
+        std::uint32_t sum{0};
+        std::for_each(
+            filesPerType.begin(), filesPerType.end(), [&sum](const auto &record) mutable { sum += record.second; });
```

Comment:
`mutable` seems to be not needed here, also you can write this method as a oneliner using `std::accumulate`:

```
return std::accumulate(filesPerType.begin(), filesPerType.end(), 0, [](const auto sum, const auto &item) { return sum + item.second; });
```

Reply:
Thanks, I didn't know this function :D